test/adapters/generic_adapter_test.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class GenericAdapterTest < ActiveSupport::TestCase
+  test 'new' do
+    assert GenericAdapter.new('http://id:secret@lvh.me:3000/auth/realm/name')
+  end
+
+  test 'endpoint' do
+    adapter = GenericAdapter.new('http://id:secret@lvh.me:3000/auth/realm/name')
+
+    assert_kind_of URI, adapter.endpoint
+  end
+
+  test 'setting access token' do
+    subject = GenericAdapter.new('http://lvh.me:3000')
+
+    subject.authentication = 'sometoken'
+
+    assert_equal 'sometoken', subject.authentication
+  end
+
+  test 'endpoint normalization' do
+    uri = URI('http://lvh.me:3000/auth/realm/name/')
+
+    assert_equal uri,
+                 GenericAdapter.new('http://id:secret@lvh.me:3000/auth/realm/name').endpoint
+
+    assert_equal uri,
+                 GenericAdapter.new('http://id:secret@lvh.me:3000/auth/realm/name/').endpoint
+  end
+
+  test 'timeout error' do
+    stub_request(:get, 'http://lvh.me:3000/auth/realm/name/.well-known/openid-configuration').
+        to_return(status: 200, headers: { 'Content-Type' => 'application/json' },
+                  body: { token_endpoint: 'protocol/openid-connect/token' }.to_json)
+
+    stub_request(:post, 'http://lvh.me:3000/auth/realm/name/protocol/openid-connect/token').to_timeout
+
+    adapter = GenericAdapter.new('http://id:secret@lvh.me:3000/auth/realm/name')
+
+    begin
```

Comment:
Without begin/rescue

```ruby
error = assert_raise GenericAdapter::OIDC::AuthenticationError do
  adapter.test
end

assert_kind_of Faraday::TimeoutError, error.cause
assert error.bugsnag_meta_data.presence
```

Reply:
I'll revert your d34e9d1. Feel free to apply those changes when they are working. But I need to merge & deploy this.

Reply:
Fair enough, I will work on it